torchtune/data/_chat_formats.py

```diff
@@ -0,0 +1,215 @@
+
+    system = f"{B_SYS}{{content}}{E_SYS}"
+    user = f"{B_INST} {{system_message}}{{content}} {E_INST} "
+    assistant = ""
+
+    @classmethod
+    def format(
+        cls,
+        sample: List[Message],
+    ) -> List[Message]:
+        """
+        Format user and system messages with appropriate tags.
+
+        Args:
+            sample (List[Message]): a single conversation, structured as a list
+                of `Message` objects
+
+        Returns:
+            The formatted list of messages
+        """
+        system_message = ""
+        formatted_dialogue = []
+        for message in sample:
+            content = ""
+            if message.role == "system":
+                content = cls.system.format(content=message.content)
+                system_message = content
+                # Incorporate the system message in the user message - LLaMA2 only
+                # looks for the <<SYS>> tags and not the explicit role so this will
+                # be treated the same as an actual system message. We do this because
+                # of the nesting of the system prompt in the user message.
+                continue
+            elif message.role == "user":
+                content = cls.user.format(
+                    system_message=system_message, content=message.content
+                )
+            elif message.role == "assistant":
+                # No special formatting needed for assistant message
+                content = message.content
+            assert content != ""
+            formatted_dialogue.append(
+                Message(role=message.role, content=content, masked=message.masked),
+            )
+        return formatted_dialogue
+
+
+class MistralChatFormat(ChatFormat):
+    """
+    Formats according to Mistral's instruct model:
+    https://docs.mistral.ai/models/
+
+    It is identical to `Llama2ChatFormat`, except it does not support system
+    prompts.
+
+    Example:
+        "[INST] I am going to Paris, what should I see? [/INST] Paris, the capital
+        of France, is known for its stunning architecture..."
+
+    """
+
+    B_INST, E_INST = "[INST]", "[/INST]"
+    system = ""
+    user = f"{B_INST} {{content}} {E_INST} "
+    assistant = ""
+
+    @classmethod
+    def format(
+        cls,
+        sample: List[Message],
+    ) -> List[Message]:
+        """
+        Format user and system messages with appropriate tags.
+
+        Args:
+            sample (List[Message]): a single conversation, structured as a list
+                of `Message` objects
+
+        Returns:
+            The formatted list of messages
+
+        Raises:
+            ValueError: If system prompts are provided
+        """
+        formatted_dialogue = []
+        for message in sample:
+            content = ""
+            if message.role == "system":
+                raise ValueError(
+                    "System prompts are not supported in MistralChatFormat"
+                )
```

Comment:
Sorry for the very noob question, but why is this not a pass through as we do for assistant? Basically, are we saying that this format doesnt support "system" as a role or that it doesn't support the system tags? Or are they the same thing? And if "system" isn't supported then should we be setting system above to None instead of empty string?

Reply:
@RdoubleA may have the best answer here tbh

Reply:
Mistral does not support the system role, so if a user passes in a message with a system role, we need to either error out or raise a warning that it will be ignored. See for context: https://github.com/vllm-project/vllm/issues/2080#issuecomment-1854436854

Agreed on setting system to None instead of an empty string.